### DISCUSSION TITLE
Main branch is not detected as main branch and produces pre-release versions

### DISCUSSION THREAD

**Discussion by dp-sgr** [GitTools/GitVersion#4218] [Q&A] [answered]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/GitTools/GitVersion/issues) to ensure it has not already been reported

### GitVersion package

GitVersion.Tool

### GitVersion version

6.0.2

### Operating system

Windows

### What are you seeing?

In earlier GitVersion releases we had the following workflow:

develop: produces alpha releases
release/*: produces beta releases
main: Commits on main always lead to full RTM releases. We used GitVersion to automatically tag back to the repo the newly created versions in our CI pipeline.

Version incremention strategies are made through:

- Patches simply merge the PR on main
- Minor/Major with either the merge from a relase branch or +semver: (major|minor)

This was achieved without any specific GitVersion config. We always used the standard.

### What is expected?

I expect that every commit to the main branch is detected as RTM and not a prerelease version.

Regardless of new repos. This also doesn't  work anymore on older repos with a big versioning history.

It seems like the main branch is not detected properly. (look at screenshots in "Steps to Reproduce")

### Steps to Reproduce

1. Create a new git repo with a main branch
2. Commit a file (e.g. README)
3. Run GitVersion

OR:
Pull my [Test Repo](https://github.com/dp-sgr/GitVersion_MainIncrementBug) and run GitVersion.

Here is the output of gitversion /showconfig
<img width="225" alt="image" src="https://github.com/user-attachments/assets/fe41c7a6-4e91-47d4-9455-4b59b1fb5ef6">

Which currently leads to the following versioning output:

``` json
{
  "AssemblySemFileVer": "0.0.1.0",
  "AssemblySemVer": "0.0.1.0",
  "BranchName": "main",
  "BuildMetaData": null,
  "CommitDate": "2024-09-19",
  "CommitsSinceVersionSource": 1,
  "EscapedBranchName": "main",
  "FullBuildMetaData": "Branch.main.Sha.0ab9e5efc330761e8bb359220505da80e337a0b8",
  "FullSemVer": "0.0.1-1",
  "InformationalVersion": "0.0.1-1+Branch.main.Sha.0ab9e5efc330761e8bb359220505da80e337a0b8",
  "Major": 0,
  "MajorMinorPatch": "0.0.1",
  "Minor": 0,
  "Patch": 1,
  "PreReleaseLabel": "",
  "PreReleaseLabelWithDash": "",
  "PreReleaseNumber": 1,
  "PreReleaseTag": "1",
  "PreReleaseTagWithDash": "-1",
  "SemVer": "0.0.1-1",
  "Sha": "0ab9e5efc330761e8bb359220505da80e337a0b8",
  "ShortSha": "0ab9e5e",
  "UncommittedChanges": 0,
  "VersionSourceSha": "",
  "WeightedPreReleaseNumber": 55001
}
```

As you can see SemVer/FullSemVer is now 0.0.1-1 and i get a PreReleaseTag. 
I expect that the output of this results in "0.0.1" without any prerelease tag as in prior GitVersion releases.

Is this a bug or do i need to explecitly configure this behavior with Gitversion 6?
If this is a configuration problem can you please tell me what i need to adjust?

**Comment by HHobeck** [comment, marked as answer]
Please create only an issue if you know you have one. Also before posting it would be very decent to read the documentation at https://gitversion.net/.

[Continuous Deployment](https://gitversion.net/docs/reference/modes/continuous-deployment)

> Continuous Deployment is the process of checking into main and automatically deploying to production.
> 
> For this mode we follow the logic in [this blog post by Xavier Decoster](https://www.xavierdecoster.com/semantic-versioning-auto-incremented-nuget-package-versions) on the issues of incrementing automatically.
> 
> How Continuous Deployment affects GitVersion
> The thing about continuous deployment is that there will be only one version to deploy on production. This means that GitVersion will build the same semantic version for every commit until it has been tagged. For instance:
> 
> 1.2.0
> 1.1.0 (tag: 1.1.0) <-- This is the version which has been deployed on production
> 1.1.0
> 1.1.1
> Tags are required in this mode to communicate when the deployement happens on production.

[GitFlow Examples](https://gitversion.net/docs/learn/branching-strategies/gitflow/examples)
> The continuous delivery mode has been used for the main and the support branch in this examples (specified as a fallback on the root configuration layer) to illustrate how the version increments are applied. In production context the continuous deployment mode might be a better option when e.g. the release process is automated or the commits are tagged by the pipeline automatically.

and

> After the feature branch is merged, the version on main is 2.0.0-5. This is due to main running in continuous delivery mode. If main was configured to use continuous deployment the version would be 2.0.0.

Happy Branching!

Edit:

Maybe it's good to know what's changed in version 6 compared to version 5 and why it probably behaves differently:
[BREAKING_CHANGES](https://github.com/GitTools/GitVersion/blob/main/BREAKING_CHANGES.md)

> The versioning mode has been renamed to deployment mode and consists of following values:
> 
> ManualDeployment (previously ContinuousDelivery)
> ContinuousDelivery (previously ContinuousDeployment)
> ContinuousDeployment (new)

**Comment by dp-sgr** [comment]
@HHobeck Thanks for the response. Sorry for creating an issue. I thought it was an error because of the displayment of "is-main-branch: false". Didn't se its a static config value before.


If other find the issue/discussion here: Im going with the following yml for now:

```yml
branches:
  develop:
    mode: ContinuousDelivery
  main:
    mode: ContinuousDeployment
  release:
    mode: ContinuousDelivery
```

It seems to behave like correctly for my use-cases but i will need to test in the next weeks with my production repos.

**Reply by HHobeck** [reply]
Yes please test your workflow and come back if something is not clear.
